pkg/virtctl/adm/logverbosity/logverbosity.go

```diff
@@ -0,0 +1,446 @@
+		if !cmd.Flags().Changed(componentName) {
+			continue // do nothing for the component
+		}
+
+		// if flag is specified, it means either set or show
+		// if the value = noArg, it means show
+		// if the value != noArg, it means set
+		isShow = isShow || verbosities[component] == noArg
+		isSet = isSet || verbosities[component] != noArg
+
+		// check whether the verbosity is in the range
+		if verbosities[component] != noArg && verbosities[component] > maxVerbosity {
+			return nop, fmt.Errorf("%s: log verbosity must be %d-%d", componentName, minVerbosity, maxVerbosity)
+		}
+
+		// add a componentName and its verbosity to the map
+		// in case of show: verbosity = noArg
+		// in case of set: verbosity = 0-9
+		// in case of reset: no information added to the map
+		options[componentName] = verbosities[component]
+	}
+
+	// do not distinguish between set and reset at this point
+	// because set and reset can coexist
+	if isReset {
+		isSet = true
+	}
+
+	if isShow && isSet {
+		return nop, errors.New("only show or set is allowed")
+	}
+
+	if isShow {
+		return show, nil
+	} else if isSet {
+		return set, nil
+	}
+
+	return nop, nil
+}
+
+func (c *Command) RunE(cmd *cobra.Command) error {
+	// get client
+	virtClient, err := kubecli.GetKubevirtClientFromClientConfig(c.clientConfig)
+	if err != nil {
+		return err
+	}
+	// get install namespace and name
+	namespace, name, err := detectInstallNamespaceAndName(virtClient)
+	if err != nil {
+		return err
+	}
+	// get KubeVirt CR
+	kv, err := virtClient.KubeVirt(namespace).Get(name, &k8smetav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	// check the operation type (nop/show/set), and set the options map to use the map for show and set operations
+	options := map[string]uint{} // key: component name, value: verbosity
+	op, err := findOperation(cmd, options)
+	if err != nil {
+		return err
+	}
+
+	switch op {
+	case nop:
+		err := cmd.Help()
+		if err != nil {
+			return err
+		}
+		return errors.New("no flag specified - expecting at least one flag")
+	case show:
+		// if verbosity has been set in the KubeVirt CR, use the verbosity
+		currentLv, err := hasVerbosityInKV(kv)
+		if err != nil {
+			return err
+		}
+		lines := createShowMessage(currentLv, options)
+		for _, line := range lines {
+			cmd.Println(line)
+		}
+	case set: // set and/or reset
+		// if there is a logVerbosity field in the KubeVirt CR, fill the verbosity in the map
+		// when the verbosity is not specified for the component, and there is an existing verbosity in KubeVirt CR,
+		// we need currentLv (the existing verbosity in the KubeVirt CR),
+		// because if we do not use the existing verbosity, the existing verbosity will be removed.
+		currentLv, err := hasVerbosityInKV(kv)
+		if err != nil {
+			return err
+		}
+		// if map is nil (= logVerbosity field in the KubeVert CR is nil), need initialization to access a key in the map
+		if currentLv == nil {
+			currentLv = make(map[string]uint)
+		}
```

Comment:
As mentioned earlier, I think we can make `hasVerbosityInKV()` always return a valid `map[string]uint` so this check is not needed.

Reply:
As answered earlier